instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go

```diff
@@ -60,12 +60,15 @@ func Middleware(service string, opts ...Option) gin.HandlerFunc {
 			c.Request = c.Request.WithContext(savedCtx)
 		}()
 		ctx := cfg.Propagators.Extract(savedCtx, propagation.HeaderCarrier(c.Request.Header))
-		opts := []oteltrace.SpanStartOption{
-			oteltrace.WithAttributes(semconv.NetAttributesFromHTTPRequest("tcp", c.Request)...),
-			oteltrace.WithAttributes(semconv.EndUserAttributesFromHTTPRequest(c.Request)...),
-			oteltrace.WithAttributes(semconv.HTTPServerAttributesFromHTTPRequest(service, c.FullPath(), c.Request)...),
-			oteltrace.WithSpanKind(oteltrace.SpanKindServer),
-		}
+		opts := cfg.SpanStartOptions
+		opts = append(opts,
+			[]oteltrace.SpanStartOption{
+				oteltrace.WithAttributes(semconv.NetAttributesFromHTTPRequest("tcp", c.Request)...),
+				oteltrace.WithAttributes(semconv.EndUserAttributesFromHTTPRequest(c.Request)...),
+				oteltrace.WithAttributes(semconv.HTTPServerAttributesFromHTTPRequest(service, c.FullPath(), c.Request)...),
+				oteltrace.WithSpanKind(oteltrace.SpanKindServer),
```

Comment:
This order of append means the default options always get precedence. So it's impossible to override the span kind. That may be fine though.

Reply:
is it necessary to support overwrite otelgin settings, like span kind and some http related attributes ?